cm/lib/CMReconstruction.cc

```diff
@@ -146,7 +147,59 @@ void CMReconstruction::RunReconstruction(Int_t nIterations) {
     throw "too many iterations";
   }
 
-  FillHMatrix();
+
+  if(CmdLineOption::GetStringValue("Hmatrix")){ 
+    TString hfilename(CmdLineOption::GetStringValue("Hmatrix"));
+    log->info("Hmatrix file: {}",hfilename);
+    TFile hfile(hfilename);
+    hfile.cd();
+
+    fMatrixH.Read("matrixH");
+    fMatrixHPrime.Transpose(fMatrixH);
+    Mask fMaskCheck = *static_cast<Mask*>(hfile.Get("mask"));
+    DetPlane fDetPlaneCheck = *static_cast<DetPlane*>(hfile.Get("detector"));
+
+    H2Coords fMaskCheckCoords(fMaskCheck.GetPattern());
+    H2Coords fMaskCoords(fMask.GetPattern());
+    Int_t fMaskCheckBins = fMaskCheckCoords.NBins();
+    Int_t fMaskBins = fMaskCoords.NBins();
+    log->info("Mask1 bins: {}",fMaskCheckCoords.NBins());
+    log->info("Mask2 bins: {}",fMaskCoords.NBins());
+
+    Double_t ma1 = fMaskCheck.GetA(), mb1 = fMaskCheck.GetB(), mc1 = fMaskCheck.GetC();
+    Double_t md1 = fMaskCheck.GetD(), mY1 = fMaskCheck.GetDimY(), mZ1 = fMaskCheck.GetDimZ();
+    Double_t da1 = fDetPlaneCheck.GetA(), db1 = fDetPlaneCheck.GetB(), dc1 = fDetPlaneCheck.GetC();
+    Double_t dd1 = fDetPlaneCheck.GetD(), dY1 = fDetPlaneCheck.GetDimY(), dZ1 = fDetPlaneCheck.GetDimZ();
+
+    Double_t ma2 = fMask.GetA(), mb2 = fMask.GetB(), mc2 = fMask.GetC();
+    Double_t md2 = fMask.GetD(), mY2 = fMask.GetDimY(), mZ2 = fMask.GetDimZ();
+    Double_t da2 = fDetPlane.GetA(), db2 = fDetPlane.GetB(), dc2 = fDetPlane.GetC();
+    Double_t dd2 = fDetPlane.GetD(), dY2 = fDetPlane.GetDimY(), dZ2 = fDetPlane.GetDimZ();
+
+    //TODO: separate conditions in order to recognize reason of error
```

Comment:
Just finish it and add these separate checks, otherwise it might never be done later :-)